### ISSUE TITLE
2.0.0 possible regression: HtmlHeadDecorator now requires a source <head/>

### ISSUE THREAD

**Issue by makigumo** [ultraq/thymeleaf-layout-dialect#113]
Previously it would be ok to completely omit an (empty) `<head>` element in e.g.:

```html
 <html xmlns:th="http://www.thymeleaf.org"
       xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
       layout:decorator="common">
 <section layout:fragment="content" id="content">
 [..]
 </section>
 </html>
```

In the latest snapshots this fails with the following exception.

```
java.lang.NullPointerException: Cannot invoke method findModel() on null object
[..]
	at nz.net.ultraq.thymeleaf.decorators.html.HtmlHeadDecorator.decorate(HtmlHeadDecorator.groovy:70)
```

This regression was probably introduced with commit 336754f. A quick look shows that there is no check for `null` in `sourceHeadModel`, which is provided by the `sourceDocumentModel.findModel(headModelFinder)` call in [`HtmlDocumentDecorator.groovy:66`](https://github.com/ultraq/thymeleaf-layout-dialect/blob/thymeleaf-3.0/Source/nz/net/ultraq/thymeleaf/decorators/html/HtmlDocumentDecorator.groovy#L66).

**Comment by ultraq** [comment]
Hi @makigumo, as you've noticed, this is a regression in the the 2.0.0-SNAPSHOT and I can tell you now that those specific tests are off because I know it doesn't work! 😝

I do aim to have this resolved before a proper 2.0.0 release, but I'll keep this issue open so that you can be notified when this particular issue is available in a snapshot version for you to pick up.

**Comment by ultraq** [comment]
I've made some updates that should help this issue, restored all the relevant tests, made another fix or 2, and the tests seem to be passing! :)  I've uploaded a new `2.0.0-SNAPSHOT` with these fixes, so that should be this regression bug squashed.
